contents/rust/Rust.md

```diff
@@ -0,0 +1,79 @@
+
+There are also a lot of other details that go into making a great developer experience, such as tooling and documentation. However, instead of going into more details, let's look at a resource that summarizes why others think Rust is great:
+
+[fireflowers - The Rust Programming Language, in the words of its practitioners](https://brson.github.io/fireflowers/)
+
+## How to learn Rust
+
+Knowing the merits of a language without knowing how to learn it however is not sufficient. As [mentioned](#developer-experience), part of making the developer experience great is good documentation. And there are many great avenues to learn Rust.
+
+### Rust is difficult
+
+There is this notion however that Rust is difficult that we need to address first.
+
+It is true that there are many who say that Rust is difficult, and we should certainly acknowledge it. So while we cannot then pretend that the language is easy, there has however since been many [improvements](https://blog.rust-lang.org/2017/12/21/rust-in-2017.html#rust-should-have-a-lower-learning-curve) to reduce the learning curve.
+
+### Resources to learn Rust
+
+There are already great official resources to learn Rust, for instance:
+- [The Rust Programming Language book](https://doc.rust-lang.org/book/)
+- [Rust By Example](https://rustbyexample.com/)
+- [Frequently Asked Questions](https://www.rust-lang.org/en-US/faq.html)
+
+There is also a [large collection of community-maintained resources](https://github.com/ctjhoa/rust-learning), and the following are some resources that are possibly useful:
+* [Comparison with other languages](https://github.com/ctjhoa/rust-learning#comparison-with-other-languages)
+* [Cheat sheets](https://github.com/ctjhoa/rust-learning#cheat-sheets)
+* [Ownership / Concurrency](https://github.com/ctjhoa/rust-learning#ownership--concurrency)
+
+More resources: [Rust Official Documentation](https://www.rust-lang.org/en-US/documentation.html)
+
+# Why not Rust
+
+Choosing the right language for the job is important so let's look at a few possible reasons not to use Rust:
+- [The ecosystem is not as mature](https://github.com/ctjhoa/rust-learning#are-we--yet)
+- Rust's compiler is extremely strict in that it enforces ownership, i.e. needs time getting used to. However, it is [probably debatable if this approach is a worthwhile trade-off](https://news.ycombinator.com/item?id=16202373).
+- While not a completely similar language, [Go feels immediately more productive](https://news.ycombinator.com/item?id=13430108)
+
+Some other discussions on why not Rust:
+- https://www.reddit.com/r/rust/comments/7p75ab/why_rust_what_i_want_changed_for_rust_to_help_way/
```

Comment:
- [How can Rust improve](https://www.reddit.com/r/rust/comments/7p75ab/why_rust_what_i_want_changed_for_rust_to_help_way/)
- [Why not Rust](https://www.reddit.com/r/rust/comments/6hp54n/blog_why_not_to_use_rust/)